src/x86/sse41.rs

```diff
@@ -233,6 +249,116 @@ pub unsafe fn _mm_dp_ps(a: f32x4, b: f32x4, imm8: u8) -> f32x4 {
     constify_imm8!(imm8, call)
 }
 
+// TODO docs
+#[inline(always)]
+#[target_feature = "+sse4.1"]
+#[cfg_attr(test, assert_instr(roundpd))]
+pub unsafe fn _mm_floor_pd(a: f64x2) -> f64x2 {
+    roundpd(a, _MM_FROUND_FLOOR)
+}
+
+// TODO docs
+#[inline(always)]
+#[target_feature = "+sse4.1"]
+#[cfg_attr(test, assert_instr(roundps))]
+pub unsafe fn _mm_floor_ps(a: f32x4) -> f32x4 {
+    roundps(a, _MM_FROUND_FLOOR)
+}
+
+// TODO docs
+#[inline(always)]
+#[target_feature = "+sse4.1"]
+#[cfg_attr(test, assert_instr(roundsd))]
+pub unsafe fn _mm_floor_sd(a: f64x2, b: f64x2) -> f64x2 {
+    roundsd(a, b, _MM_FROUND_FLOOR)
+}
+
+// TODO docs
+#[inline(always)]
+#[target_feature = "+sse4.1"]
+#[cfg_attr(test, assert_instr(roundss))]
+pub unsafe fn _mm_floor_ss(a: f32x4, b: f32x4) -> f32x4 {
+    roundss(a, b, _MM_FROUND_FLOOR)
+}
+
+// TODO docs
+#[inline(always)]
+#[target_feature = "+sse4.1"]
+#[cfg_attr(test, assert_instr(roundpd))]
+pub unsafe fn _mm_ceil_pd(a: f64x2) -> f64x2 {
+    roundpd(a, _MM_FROUND_CEIL)
+}
+
+// TODO docs
+#[inline(always)]
+#[target_feature = "+sse4.1"]
+#[cfg_attr(test, assert_instr(roundps))]
+pub unsafe fn _mm_ceil_ps(a: f32x4) -> f32x4 {
+    roundps(a, _MM_FROUND_CEIL)
+}
+
+// TODO docs
+#[inline(always)]
+#[target_feature = "+sse4.1"]
+#[cfg_attr(test, assert_instr(roundsd))]
+pub unsafe fn _mm_ceil_sd(a: f64x2, b: f64x2) -> f64x2 {
+    roundsd(a, b, _MM_FROUND_CEIL)
+}
+
+// TODO docs
+#[inline(always)]
+#[target_feature = "+sse4.1"]
+#[cfg_attr(test, assert_instr(roundss))]
+pub unsafe fn _mm_ceil_ss(a: f32x4, b: f32x4) -> f32x4 {
+    roundss(a, b, _MM_FROUND_CEIL)
+}
+
+// TODO docs
+#[inline(always)]
+#[target_feature = "+sse4.1"]
+#[cfg_attr(test, assert_instr(roundpd))]
```

Comment:
When testing intrinsics with const immediates, you need to generate code with a constant, otherwise you won't get the right codegen. So in this case I think something like `#[cfg_attr(test, assert_instr(roundpd, rounding = 1))]` should work. See an example here: https://github.com/rust-lang-nursery/stdsimd/blob/master/src/x86/sse2.rs#L316